owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/status/CapabilityBooleanType.kt

```diff
@@ -22,59 +22,44 @@
  *   THE SOFTWARE.
  *
  */
-package com.owncloud.android.lib.resources.status;
+package com.owncloud.android.lib.resources.status
 
 /**
- * Enum for Boolean Type in OCCapability parameters, with values:
+ * Enum for Boolean Type in RemoteCapability parameters, with values:
  * -1 - Unknown
  * 0 - False
  * 1 - True
  */
-public enum CapabilityBooleanType {
+enum class CapabilityBooleanType private constructor(val value: Int) {
     UNKNOWN(-1),
     FALSE(0),
     TRUE(1);
 
-    private int value;
+    val isUnknown: Boolean
+        get() = value == -1
 
-    CapabilityBooleanType(int value) {
-        this.value = value;
-    }
+    val isFalse: Boolean
+        get() = value == 0
 
-    public static CapabilityBooleanType fromValue(int value) {
-        switch (value) {
-            case -1:
-                return UNKNOWN;
-            case 0:
-                return FALSE;
-            case 1:
-                return TRUE;
-        }
-        return null;
-    }
+    val isTrue: Boolean
+        get() = value == 1
 
-    public static CapabilityBooleanType fromBooleanValue(boolean boolValue) {
-        if (boolValue) {
-            return TRUE;
-        } else {
-            return FALSE;
+    companion object {
+        fun fromValue(value: Int): CapabilityBooleanType? {
```

Comment:
the same as before. Include a else condition in the when block for null values

Reply:
Done